rfc/rfc-0020-resources-update-after-success.md

```diff
@@ -0,0 +1,225 @@
+
+Let us again consider resource A. Two updates are made to resource A in quick succession. The first succeeds and
+updates the `latestGoodOutput`. But the status is still Ready:unknown, so Cartographer does not propagate this output.
+The second fails and results in `Ready:false`. Cartographer will not propagate any output because `Ready:false`.
+How can this be addressed?
+
+#### Insufficient Strategy
+
+Cartographer could wait to submit a new spec until the previous spec has finished reconciling. This would still need
+to be paired with only reading when Ready:true (as demonstrated above). But even more than that we can see that this will
+either leave Carto vulnerable to an equivalent problem or introduce potential bottlenecks in the supply chain. The
+difference is in the method of submitting waiting updates.
+a. Assume a strategy of not submitting while work is occurring, when work is done, submit the most recent update
+(pop from the stack). Here we find a problem analogous to the original. Consider update 1. While it works, update 2
+(good) and 3 (bad) come in. Once update 1 finishes processing, update 3 will be submitted. Update 3, being bad, does
+not result in a valid output. And the latest update that would produce an output (update 2) is never submitted. Bad Carto.
+b. Again assume a strategy of not submitting while work is occurring. This time, when work is done, submit the earliest
+update (first in the queue). This will result in every update being submitted. But it will be slow. If resource A takes
+a long time to reconcile, users may not want to wait on outdated updates.
+
+#### Happy Middle Ground Strategy
+
+As mentioned, strategy A pops the most recent update from a stack. This suggests an optimal strategy. Pop updates from
+the stack until an update produces a good output. At that point the stack can be cleared. This assures that the most
+recent good update results in an output, but does not waste time needlessly calculating the output of updates that have
+already been superseded.
+
+## Implementation details
+
+### Use ObservedCompletion and ObservedMatch (from DeploymentTemplate)
+
+Templates that expose outputs (SourceTemplate, ImageTemplate...) can include success conditions. These can include
+expected values at a given path on the stamped object. These can alternately be an expectation that one field (e.g.
+on the status) matches another field (e.g. on the spec).
+
+- An observed completion: includes a mandatory SucceededCondition and an optional FailedCondition. Both conditions are 
+  defined by a path and value. When the object's observedGeneration == generation, and the value at the
+  specified path matches the stated value, then this condition is met. 
+- An observed matches: a list of matches. Each match is a definition of two paths. When the values at the two paths are
+  the same, then this condition is met. This can be used for resources that do not report observedCondition, but whose
```

Comment:
Do you mean `observedGeneration`?
```suggestion
  the same, then this condition is met. This can be used for resources that do not report observedGeneration, but whose
```

Reply:
I did. Thank you!